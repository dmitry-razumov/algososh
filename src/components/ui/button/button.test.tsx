import { Button } from "./button"
import { fireEvent, render, screen } from "@testing-library/react";

describe ('Button component tests', () => {
  test('Should render button with text', () => {
    render(<Button text="someText"/>);
    expect(screen.getByRole('button')).toMatchSnapshot();
  })

  test('Should render button without text', () => {
    render(<Button />);
    expect(screen.getByRole('button')).toMatchSnapshot();
  })

  test('Should be disabled', () => {
    render(<Button disabled={true}/>);
    expect(screen.getByRole('button')).toMatchSnapshot();
  })

  test('Should be with loading', () => {
    render(<Button isLoader={true}/>);
    expect(screen.getByRole('button')).toMatchSnapshot();
  })

  test('Should be called callback 1 times', () => {
    const onClickCallback = jest.fn();
    render(<Button onClick={onClickCallback}/>);
    fireEvent.click(screen.getByRole('button'));
    expect(onClickCallback).toBeCalledTimes(1);
  })
})