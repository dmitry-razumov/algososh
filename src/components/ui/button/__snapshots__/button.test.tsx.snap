// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests Should be disabled 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component tests Should be with loading 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component tests Should render button with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    someText
  </p>
</button>
`;

exports[`Button component tests Should render button without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
